trigger/project_triggers.go

```diff
@@ -41,20 +42,28 @@ func TriggerDownstreamVersion(ctx context.Context, args ProcessorArgs) (*model.V
 	if err != nil {
 		return nil, errors.Wrap(err, "creating version")
 	}
-	err = args.SourceVersion.AddSatisfiedTrigger(args.DefinitionID)
-	if err != nil {
-		return nil, err
+	if args.SourceVersion != nil {
+		err = args.SourceVersion.AddSatisfiedTrigger(args.DefinitionID)
+		if err != nil {
+			return nil, err
+		}
 	}
 	settings, err := evergreen.GetConfig(ctx)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting Evergreen settings")
 	}
-	upstreamProject, err := model.FindMergedProjectRef(args.SourceVersion.Identifier, args.SourceVersion.Id, true)
+	var versionID string
+	projectID := args.ProjectID
+	if projectID == "" {
```

Comment:
Sorry if I didn't catch this earlier but this change is a bit confusing to me; wouldn't we want to assign versionID to args.SourceVersion.Id unless it's empty? Where does args.ProjectID come from that we wouldn't want to overwrite it?

Reply:
Might be worth adding some comments here so we remember this special case

Reply:
Sure, I added a clarifying comment